src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -158,6 +158,11 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_copy_mem_reg,
     CUresult result;
     ucs_status_t status;
 
+    if (!uct_cuda_base_is_context_active()) {
+        ucs_debug("attempt to register memory without active context");
+        return uct_md_dummy_mem_reg(md, address, length, params, memh_p);
+    }
```

Comment:
does it fix the problem whith that shmem app (i guess with perf penalty then)?
btw, why did we remove it in 1.16  - i see the same code is present  in master

Reply:
It should fix the issue. UCX tries to register host memory using all available MDs. It was agreed that we should do dummy memory registration when there is no active CUDA context (aka CUDA device is not set).
1.16 branch was created before https://github.com/openucx/ucx/pull/8590 was merged to master branch.